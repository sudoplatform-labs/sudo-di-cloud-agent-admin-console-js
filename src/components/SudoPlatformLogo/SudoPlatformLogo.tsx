import React from 'react';
import styled from 'styled-components';
import { Image, ImageName } from '../Image';

const SudoPlatformLogoContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledImage = styled(Image)<{ width?: number }>`
  display: block;
  width: ${(props) => `${props.width}%` ?? 'auto'};
  height: auto;
  margin-left: 18px;
`;

interface Props {
  className?: string;
  name: ImageName;
  width?: number;
}

export const SudoPlatformLogo = (props: Props): JSX.Element => (
  <SudoPlatformLogoContainer className={props.className}>
    <StyledImage name={props.name} width={props.width} />
  </SudoPlatformLogoContainer>
);
