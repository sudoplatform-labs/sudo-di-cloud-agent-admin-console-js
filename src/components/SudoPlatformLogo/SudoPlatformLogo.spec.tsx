import { shallow } from 'enzyme';
import React from 'react';
import { SudoPlatformLogo } from './SudoPlatformLogo';

describe('BreadCrumbs', () => {
  it('should render correctly', () => {
    const component = shallow(<SudoPlatformLogo name="sdca_light" />);

    expect(component).toMatchSnapshot();
  });
});
