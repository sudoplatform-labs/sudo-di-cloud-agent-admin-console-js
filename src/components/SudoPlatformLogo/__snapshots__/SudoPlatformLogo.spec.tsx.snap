// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs should render correctly 1`] = `
<styled.div>
  <Styled(Image)
    name="sdca_light"
  />
</styled.div>
`;
